design/carvel-install-design.md

```diff
@@ -0,0 +1,54 @@
+
+## Requirements
+### Organization
+- The generated Velero configuration templates and artifacts must be kept in the Velero GitHub repository.
+- The confiruation artifacts must be be laid out in a directory structure in a way that conforms with the expected standard so that they can be easily discovered by configuration consumers.
+
+### Configuration
+- The creation and configuration of Velero deployment and other non-CRD resources must be migrated from being generated using Go code to being available as artifacts that the Carvel tooling can consume to create and update these resources in Kubernetes clusters.
+- Pursue an "ask once" approach to configuring the Velero deployment. In other words: given the custom configuration input from the configuration consumer, independely determine what environment variables and defaults, and other configuration settings are necessary for that setup.
+- A default Backup Storage Location must be automatically configured and created based on the custom values provided by the configuration consumer.
+- For compatibility with devops, Velero must continue to provide a `dry-run` option via the Carvel tooling for outputing configuration artifacts without installing them.
+- Because the Carvel tooling, given proper templates, offers full functionality to do what the current Velero install command does, namelly create deployment and other resources in Kubernetes clusters, as well as update them, a path to deprecating that command could be considered and defined. An intermediate step might be to wrap the `install` command around the Carvel tooling.
+- Documentation and sample templates to be used by the configuration consumers must be provided for each environment supported.
+
+### Release packaging
+- The resulting artifacts, including CRDs, must be packaged so that it contains all the appropriate set of configuration that comprises a Velero release.
+- Users must be able to upgrade and downgrade Velero versions.
+- The Velero/Carvel integration will be supported with the Velero v1.7 and onwards.
+- Upgrades from and downgrades to previous versions of Velero must be complatible with the v1.7 way of installation.
+
+
+## Supported environments
+Envirornments that need to be supported: AWS, Vsphere, Azure, Kind.
+
+## Configuration to be supported per enviroment
+
+| Configuration                              |      AWS       |   Vsphere  |  Azure    | Kind  |
+| ------------------------------------------ | -------------- | ---------- | ----------| ----- |
+| Backup/restore of K8s resources            |      ✅        |     ✅     |    ✅      |   ✅  |
+| Backup/restore of PVs via plugins          |      ✅        |     ✅     |    ✅      |  N/A  |
+| Backup/restore of PVs with restic          |      ✅        |     ✅     |    ✅      |  N/A  |
+
```

Comment:
For plug-ins, do we want to be able to specify plug-in versions separate from Velero versions or do we wrap all of that together in a single "release" and create a new release if we have a Velero or plug-in update?